storage/tests/system.py

```diff
@@ -414,29 +416,52 @@ def test_crud_blob_w_user_project(self):
 
         # Exercise 'objects.insert' w/ userProject.
         blob.upload_from_filename(file_data["path"])
+        gen0 = blob.generation
+
+        # Upload a second generation of the blob
+        blob.upload_from_file(io.StringIO(six.text_type("gen1")))
```

Comment:
This should be `io.BytesIO`, rather than `io.StringIO`.  Or you could just use `blob.upload_from_string(b"gen1")`.